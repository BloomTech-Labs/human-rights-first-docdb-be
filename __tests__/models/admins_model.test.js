const db = require('../../data/db-config');
const Admins = require('../../api/admin/adminModel');

beforeAll(async () => {
  await db.migrate.rollback();
  await db.migrate.latest();
});

beforeEach(async () => {
  await db.seed.run();
});

afterAll(async () => {
  await db.destroy();
});

describe('findAll()', () => {
  let res;
  beforeEach(async () => {
    res = await Admins.findAll();
  });
  it('returns 2 admins', () => {
    expect(res).toHaveLength(2);
  });
  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});

describe('findById(adminId)', () => {
  let res;
  const adminId = 23423;
  beforeEach(async () => {
    res = await Admins.findById(adminId);
  });
  it('returns correct profile by findById', () => {
    expect(res.name).toBe('John Smith');
    expect(res.id).toBe('abc123');
  });
  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});

describe('create(newAdmin)', () => {
  let res;
  const newAdmin = {
    id: 'j3iasd',
  };
  const newlyCreatedAdmin = {
    adminId: 1,
    id: 'j3iasd',
    name: 'Brian Cranston',
    email: 'brian@gmail.com',
  };

  beforeEach(async () => {
    res = await Admins.create(newAdmin);
  });
  it('returns newly created admin', () => {
    expect(res.name).toBe(newlyCreatedAdmin.name);
    expect(res.email).toBe(newlyCreatedAdmin.email);
  });
  it('returns 3 admins', async () => {
    const getAll = await db('admins');
    expect(getAll).toHaveLength(3);
  });
  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});
