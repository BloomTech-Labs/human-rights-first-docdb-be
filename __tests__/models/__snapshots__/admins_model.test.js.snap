// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create(newAdmin) returns the correct data shape 1`] = `
Object {
  "adminId": 1,
  "email": "brian@gmail.com",
  "id": "j3iasd",
  "name": "Brian Cranston",
}
`;

exports[`findAll() returns the correct data shape 1`] = `
Array [
  Object {
    "adminId": 3424,
    "email": "mary@gmail.com",
    "id": "2h38dh",
    "name": "Mary Berry",
  },
  Object {
    "adminId": 23423,
    "email": "abc@gmail.com",
    "id": "abc123",
    "name": "John Smith",
  },
]
`;

exports[`findById(adminId) returns the correct data shape 1`] = `
Object {
  "adminId": 23423,
  "email": "abc@gmail.com",
  "id": "abc123",
  "name": "John Smith",
}
`;
